internal/backend/k8s.go

```diff
@@ -212,3 +217,31 @@ func (c *K8sPersister) ListAccessBindings(ctx context.Context, roleName, namespa
 	}
 	return list, nil
 }
+
+func (c *K8sPersister) GetApplication(ctx context.Context, name, namespace string) (*unstructured.Unstructured, error) {
+	obj := &unstructured.Unstructured{}
+	obj.SetGroupVersionKind(v1alpha1.ApplicationGroupVersionKind)
+	key := client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+	err := c.client.Get(ctx, key, obj)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving application %s/%s from k8s: %w", namespace, name, err)
+	}
+	return obj, nil
+}
+
+func (c *K8sPersister) GetAppProject(ctx context.Context, name, namespace string) (*unstructured.Unstructured, error) {
```

Comment:
same comment for `k8s.GetApplication()` applies.

Reply:
All godocs added on the interface type